src/main/java/edu/ucar/unidata/rosetta/service/EmbeddedDerbyDbInitManager.java

```diff
@@ -0,0 +1,218 @@
+                    "templateFileName VARCHAR(100), " +
+                    "headerLineNumbers VARCHAR(100), " +
+                    "noHeaderLines VARCHAR(10), " +
+                    "delimiter VARCHAR(50)" +
+                    ")";
+            createTable(createDataTable, props);
+
+            // Populate properties table.
+            populatePropertiesTable(props);
+
+            // Okay, we're done.  Shut down this particular connection to the database.
+            try {
+                connection = DriverManager.getConnection(url + ";shutdown=true");
+                if (connection != null) {
+                    connection.close();
+                }
+            } catch(SQLException e) {
+                // As per the Derby docs, the shutdown commands always raise SQLExceptions. (lame!)
+                logger.info("Finished creating database. Shutting down database...");
+            }
+
+
+        } else {
+            logger.info("Nothing to do here... Database already exists.");
+        }
+    }
+
+
+    /**
+     * Shuts down the embedded derby database by de-registering the driver.
+     *
+     * @param props RosettaProperties from application.properties that may be used for database shutdown.
+     * @throws SQLException   If an SQL exceptions occurs during database shutdown.
+     */
+    public void shutdownDatabase(Properties props) throws SQLException {
+
+        // Get relevant properties.
+        String rosettaHome = props.getProperty("rosetta.home");
+        String databaseName = props.getProperty("jdbc.dbName");
+        String url = props.getProperty("jdbc.url").replaceAll("\\$\\{rosetta.home}", rosettaHome);
+        url = url.replaceAll("\\$\\{jdbc.dbName}", databaseName);
+        props.setProperty("jdbc.url", url);
+
+        // Okay, we're done.  Shut down this particular connection to the database.
+        try {
+            Connection connection = DriverManager.getConnection(url + ";shutdown=true");
+            if (connection != null) {
+                connection.close();
+            }
+        } catch(SQLException e) {
+            // As per the Derby docs, the shutdown commands always raise SQLExceptions. (lame!)
+            logger.info("Shutting down database...");
+        }
+
+
+        Driver driver = DriverManager.getDriver(url);
+        logger.info("De-registering jdbc driver.");
+        DriverManager.deregisterDriver(driver);
+    }
+
+    /**
+     * Creates a table in the derby database.
+     *
+     * @param statement The create SQL statement.
+     * @param props     RosettaProperties from which the database username and password are glean.
+     * @throws NonTransientDataAccessResourceException  If unable to create instance of the database driver.
+     * @throws SQLException If an SQL exceptions occurs during create table transaction.
+     */
+    private void createTable(String statement, Properties props) throws NonTransientDataAccessResourceException, SQLException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.springframework.dao.NonTransientDataAccessResourceException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)